resources/assets/js/projects/components/charts/IDToColor.js

```diff
@@ -0,0 +1,21 @@
+export function IDToColor(id) {
+    let hash = 0;
+    while (id > 0) {
+      hash += id & 0xFF;
+      hash += (hash << 10);
+      hash ^= (hash >> 6);
+      id = id >> 8;
+    }
+    hash += (hash << 3);
+    hash ^= (hash >> 11);
+    hash += (hash << 15);
+    id=hash & 0xFFFFFFFF; // Ensure it fits in 32 bits
+
+
+    // Use the hash to generate a unique hue
+    const hue = id % 360;
+    const saturation = 70 + (hash % 31); // Ensure saturation is between 70 and 100
+    const lightness = 50 + (hash % 41);  // Ensure lightness is between 50 and 90
```

Comment:
I didn't see a difference between `id` and `hash`. Did you really mean to use both here?

Reply:
Oops. Nope that was an error. Should be id all the way.